hello-world/frontend/src/App.vue

```diff
@@ -1,40 +1,110 @@
+        </p>
+        <p>
+          Expected fee: <span v-if="retrievingFee">Loading...</span>
+          <span v-else>{{ currentFee }} {{ selectedToken.symbol }}</span>
+          <button class="refresh-button" @click="updateFee">Refresh</button>
+        </p>
+      </div>
+      <div class="greeting-input">
+        <input
+          v-model="newGreeting"
+          :disabled="!selectedToken || txStatus != 0"
+          placeholder="Write new greeting here..."
+        />
+
+        <button
+          class="change-button"
+          :disabled="!selectedToken || txStatus != 0 || retrievingFee"
+          @click="changeGreeting"
+        >
+          <span v-if="selectedToken && !txStatus">Change greeting</span>
+          <span v-else-if="!selectedToken">Select token to pay fee first</span>
+          <span v-else-if="txStatus === 1">Sending tx...</span>
+          <span v-else-if="txStatus === 2"
+            >Waiting until tx is committed...</span
+          >
+          <span v-else-if="txStatus === 3">Updating the page...</span>
+          <span v-else-if="retrievingFee">Updating the fee...</span>
+        </button>
+      </div>
+    </div>
+  </div>
+  <div id="app" v-else>
+    <div class="start-screen">
+      <h1>Welcome to Greeter dApp!</h1>
+      <button v-if="correctNetwork" @click="connectMetamask">Connect Metamask</button>
+      <button v-else @click="addZkSyncSepolia">Switch to zkSync Sepolia</button>
+
+    </div>
+  </div>
+</template>
+
 <script setup lang="ts" >
-import { ref, onMounted } from 'vue'
-// TODO: import ethers and zksync-ethers
+import { ref, onMounted } from "vue"
+import { Provider, Wallet, Contract, BrowserProvider } from "zksync-ethers";
 
 const ETH_ADDRESS = "0x0000000000000000000000000000000000000000";
-import * as allowedTokens from "./eth.json"; // change to "./erc20.json" to use ERC20 tokens
+import allowedTokens from "./eth.json"; // change to "./erc20.json" to use ERC20 tokens
 
 const GREETER_CONTRACT_ADDRESS = ""; // TODO: insert the Greeter contract address here
-import * as GREETER_CONTRACT_ABI from './abi.json' // TODO: Complete and import the ABI
+import GREETER_CONTRACT_ABI from './abi.json' // TODO: Complete and import the ABI
 
 // reactive references
 const correctNetwork = ref(false)
-const tokens = ref(allowedTokens.default);
-const newGreeting:string = ref("");
-const greeting:string = ref("")
-const mainLoading:boolean = ref(true)
-const retreivingFee:boolean = ref(false)
-const retreivingBalance:boolean= ref(false)
-const currentBalance:string = ref("")
-const currentFee:string = ref("")
-const selectedTokenAddress:string = ref(null)
-const selectedToken = ref(null)
+const tokens = ref(allowedTokens);
+const newGreeting = ref("");
+const greeting = ref("")
+const mainLoading = ref(true)
+const retrievingFee = ref(false)
+const retrievingBalance = ref(false)
+const currentBalance = ref("")
+const currentFee = ref("")
+const selectedTokenAddress = ref(null)
```

Comment:
no types here @JackHamer09 ?

Reply:
There is no need to duplicate types. They are automatically derrived from passed value. Example:
- "" -> string
- true/false -> boolean

They should only be added in case of multiple types or more complex types. For example:
`const selectedTokenAddress = ref(null)` -> `const selectedTokenAddress = ref<string | null>(null)`